packages/venia-concept/src/actions/cart.js

```diff
@@ -163,21 +168,60 @@ async function fetchCartPart({ guestCartId, forceRefresh, subResource = '' }) {
     });
 }
 
-async function getGuestCartId(dispatch, getState) {
+export async function clearGuestCartId() {
+    return storage.removeItem('guestCartId');
+}
+
+export async function getGuestCartId(dispatch, getState) {
     const { cart } = getState();
 
-    // reducers may be added asynchronously
+    // ensure state slices are present
     if (!cart) {
         return null;
     }
 
-    // create a guest cart if one hasn't been created yet
     if (!cart.guestCartId) {
+        // check for a guest cart in storage
+        const storedGuestCartId = await storage.getItem('guestCartId');
+
+        // if one exists, return it
+        if (storedGuestCartId) {
+            return storedGuestCartId;
+        }
+
+        // otherwise create a guest cart
         await dispatch(createGuestCart());
     }
 
-    // retrieve app state again
+    // retrieve the new guest cart from state
     return getState().cart.guestCartId;
 }
```

Comment:
Here, we have another example of the ambiguity around async action creators.

- An async action creator is a function that receives a payload and returns a thunk
- A thunk is a function that receives `dispatch` and `getState`, performs side effects, and returns `undefined`

Here, we have a function that's essentially a naked thunk—it's not the product of a higher-order function—and it both performs side effects (using `dispatch`) and returns a value (using `getState`). Currently, it's exclusively used by async action creators, since it's of little use to components due to its shape. What should we do with this function?